go.mod

```diff
@@ -0,0 +1,8 @@
+module github.com/redhatinsights/insights-operator-conditional-gathering
+
+go 1.16
+
+require (
+	github.com/go-kit/kit v0.11.0
```

Comment:
What is the motivation for using this library pls? I am wondering if it's worth using in a fairly simple service (probably with one endpoint)

Reply:
`Go-kit` is very straightforward and organized, it provides us interoperability as well as a solid architecture design. If we want to scale this service in the future, this is the best approach. Also, it also provides our service observability, service discovery, logging, persistence, and more.

I do recommend you to read this one: https://gokit.io/faq/#why-should-i-use-go-kit

Reply:
Thanks. Well yes, but we don't need any persistence AFAIK, the observability is questionable as well, because I guess we would like to stick to our existing solution (it might be good to ask processing team here again). 

I am just saying it's always good to think and weight all the possible aspects when adding a new layer of abstraction. I also think it's good to keep things simple (this comes handy especially in case of any problems).

Reply:
I'd also go with a simpler solution. Btw, if the processing team is gonna work on the service in the future, for them it'd be new, they don't use this lib in their services. https://github.com/RedHatInsights/insights-results-smart-proxy/blob/master/go.mod

Reply:
The same goes for us...or at least for me since I don't have any experience with this lib.

Reply:
I'm sorry, but the structure used by the `insights-results-smart-proxy` is very confusing IMHO. I'm not working with the service, so, it could be the case. I think that it is too much work to spin an HTTP service TBH.

I think that if it works for them, it is fine... but I would definitely stand to use `go-kit` here. Another thing is that I would say that we probably want to use `gRPC` instead of `REST` in the future.

Even knowing that they will probably work on the service in the future, `go-kit` will not introduce anything different or super complex, instead, it provides a very clean way to develop new endpoints, with clear abstractions and layers.

Reply:
saying this - it will be more difficult to handle go-kit's based application for us. And here it seems it is not needed and the Go-way is to stick with the simplest solution with few dependencies.

Reply:
Well, actually the CVE check is a valid point... I think that is worth talking with them ('cause go-kit is a pretty nice solution for microservice development), but I think that it isn't worth it for us to wait for them. So, yep, a better solution is to remove the dependency.

Reply:
git-kit removed